PyPDF2/_page.py

```diff
@@ -365,6 +368,23 @@ def createBlankPage(
         deprecate_with_replacement("createBlankPage", "create_blank_page")
         return PageObject.create_blank_page(pdf, width, height)
 
+    @property
+    def images(self) -> List[File]:
+        images_extracted: List[File] = []
+        if RES.XOBJECT not in self[PG.RESOURCES]:  # type: ignore
+            return images_extracted
+
+        x_object = self[PG.RESOURCES][RES.XOBJECT].get_object()  # type: ignore
+        for obj in x_object:
+            if x_object[obj][IA.SUBTYPE] == "/Image":
+                mime_type, byte_stream = _xobj_to_image(x_object[obj])
+                if mime_type is not None:
+                    filename = f"{obj[1:]}.{File._mime2extension(mime_type)}"
+                    images_extracted.append(
+                        File(name=filename, data=byte_stream, mime_type=mime_type)
+                    )
```

Comment:
This strikes me as an odd abstraction, where we are passing in the `mime_type` as part of the `File` constructor, but we _also_ need to construct the full filename, using a private static function to boot, but also that the `file_extension` method doesn't correspond to the extension of the passed in `name`, but rather `mime_type`.

If we go the route of passing in the `mime_type` for the File, I'd advocate for just passing in `name` sans extension altogether and we can have a special property function that does the concatenation of name + extension to give a "filename" on demand as needed by users.

The only caveat would be for attachments, it may make sense to pass in the full filename, but I'm not well versed on that part of the spec to even know how that API might look.

Reply:
I think I'll go with 'File only has name + data (no mime_type)' for the moment, because it seems to have only advantages:

* Less clutter / less code to maintain
* No potential to discover the wrong mime type
* We could make `_xobj_to_image` just pass the file extension as before
* As `_xobj_to_image` is a private function, we can easily change the behavior if we see a clear advantage